components/partition_table/gen_esp32part.py

```diff
@@ -163,7 +164,9 @@ def __init__(self):
     def from_csv(cls, line):
         """ Parse a line from the CSV """
         line_w_defaults = line + ",,,,"  # lazy way to support default fields
-        fields = [ f.strip() for f in line_w_defaults.split(",") ]
+        def expand_var(f):
+            return os.environ[f[1:]] if f.startswith("$") else f
```

Comment:
Hi @rojer,

Sorry, this somehow fell off my radar for review.

I'm 100% OK with allowing environment variables in partition table definitions. I would prefer if we could allow mixing of variable references and plaintext in the same CSV field. Was hoping Python might have a function for this out of the box, but it looks like the best (simple) solution is a one-liner like in this SO answer: http://stackoverflow.com/a/30735354